presto-druid/src/main/java/com/facebook/presto/druid/column/SimpleReadableOffset.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.druid.column;
+
+import org.apache.druid.query.monomorphicprocessing.RuntimeShapeInspector;
+import org.apache.druid.segment.data.ReadableOffset;
+
+public class SimpleReadableOffset
+        implements ReadableOffset
+{
+    private int row;
+
+    @Override
+    public int getOffset()
+    {
+        int offset = row;
+        ++row;
```

Comment:
it's weird to increment `row` inside `getOffset`. How about having this class inheriting `Offset` rather than `ReadableOffset`?

Reply:
bad, could not inherit Offset, `BaseColumn` is requiring `ReadableOffset` in `makeColumnValuesSelector`:
`makeColumnValueSelector(ReadableOffset offset);`
let me add comment there